src/sentry/incidents/endpoints/serializers/workflow_engine_incident.py

```diff
@@ -52,49 +97,136 @@ def get_incident_status(self, priority: int | None) -> int:
+        # if we are here we have IncidentActivityType.CREATED and IncidentActivityType.DETECTED so fill those out
+        created = {
+            "id": "-1",  # TODO add lookup table to get this info
+            "incidentIdentifier": incident_identifier,
+            "type": IncidentActivityType.CREATED,
+            "value": None,
+            "previousValue": None,
+            "user": None,
+            "comment": None,
+            "dateCreated": open_period.date_started,
+        }
+        detected = created.copy()
+        detected["type"] = IncidentActivityType.DETECTED
+        open_period_activities.append(created)
+        open_period_activities.append(detected)
+
+        if open_period.resolution_activity:
+            resolved = {
                 "id": "-1",
-                "incidentIdentifier": "-1",
-                "user": {},
-                "type": 1,
-                "value": "test",
-                "previousValue": "test",
-                "comment": "test",
-                "dateCreated": datetime.now(),
+                "incidentIdentifier": incident_identifier,
+                "type": IncidentActivityType.STATUS_CHANGE,
+                "value": IncidentStatus.CLOSED,
+                "previousValue": None,
+                "user": None,
+                "comment": None,
+                "dateCreated": open_period.date_started,
             }
-        ]
+            open_period_activities.append(resolved)
 
-    def get_alert_rule(self, open_period: GroupOpenPeriod) -> AlertRuleSerializerResponse:
-        # TODO: Implement this
-        return {
-            "id": "-1",
-            "name": "Test Alert Rule",
-            "organizationId": "-1",
-            "status": 1,
-            "query": "test",
-            "aggregate": "test",
-            "timeWindow": 1,
-            "resolution": 1,
-            "thresholdPeriod": 1,
-            "triggers": [
-                {
-                    "id": "-1",
-                    "status": 1,
-                    "dateModified": datetime.now(),
-                    "dateCreated": datetime.now(),
-                }
-            ],
-            "dateModified": datetime.now(),
-            "dateCreated": datetime.now(),
-            "createdBy": {},
-            "description": "test",
-            "detectionType": "test",
+        # look up Activity rows for other status changes (warning / critical)
+        status_change_activities = Activity.objects.filter(
+            group=open_period.group, type=ActivityType.ActivityType.SET_PRIORITY
+        )
+
+        activity_status_to_incident_status = {
+            "high": IncidentStatus.CRITICAL,
+            "medium": IncidentStatus.WARNING,
         }
+        for activity in status_change_activities:
+            priority = activity.data.get("priority")
+            previous_priority = None
+            previous_activities = status_change_activities.filter(datetime__lt=activity.datetime)
```

Comment:
can we sort `status_change_activities` and walk through it to keep track of the current and previous priority instead of re-filtering it for each activity?